directives/point_of_interest.py

```diff
@@ -115,97 +120,102 @@ def run(self):
-                contentnodes.append((self.content[prevl:l-1], self.content_offset + prevl, cnode, cwidth))
-
-                colcontent = ''
-                prevl = l
-            else:
-                colcontent_l.append(batch)
-                colcontent += batch
-                colcontent += '\n'
-
-        # add last column
-        cnode = nodes.container(colcontent)
-        hcontainer.append(cnode)
-        cwidth = None
-        if colwidths:
-            cwidth = floor(colwidths[-1] * 12)
-        contentnodes.append((self.content[prevl:], self.content_offset + prevl, cnode, cwidth))
-
-        nav = nodes.container()
-        links = nodes.container()
-        # poi_nav node is later replaced by reference nodes which
-        # need to be within a TextElement-node
-        text = nodes.inline()
-        poinav = poi_nav()
-        icon = aplus_nodes.html(u'img', {
-            u'src':u'../_static/poi.png',
-            u'alt':u'Point of interest icon',
-            u'class':u'poi-icon',
-            })
-        hidelink = aplus_nodes.html(u'a', {
-            u'href':u'#' + content_name,
-            u'data-toggle':u'collapse'})
-
-        hidelink.append(icon)
-        hidelink.append(nodes.Text(title_text))
-        title.append(hidelink)
-        nav.append(title)
-        text.append(poinav)
-        links.append(text)
-        nav.append(links)
-
-        node.append(nav)
-        node.append(collapsible)
-        collapsible.append(hcontainer)
-
-        numcols = len(contentnodes)
-        for cont, offset, cnode, cwidth in contentnodes:
-            # Bootstrap uses 12 unit grid
-            if not cwidth:
-                cwidth = int(12/numcols)
-            cnode['classes'] = ['col-sm-' + str(cwidth)]
-            self.state.nested_parse(cont, offset, cnode)
-
-        # poinav id needs to be added so that we can identify the node later when processing
-        poinav['ids'].append(name)
-
-        self.options['name'] = name
-        self.add_name(node)
-        collapsible['ids'].append(content_name)
-        collapsible['classes'].extend([container_class, 'collapse'])
-
-        node['classes'].extend(['poi'])
-        if 'class' in self.options:
-            node['classes'].extend(self.options['class'])
-        title['classes'].extend(['poi-title'])
-        links['classes'].extend(['poi-links'])
-
-        if not 'hidden' in self.options:
-            collapsible['classes'].extend(['in'])
-        nav['classes'].extend([container_class])
-
-        # poi_info needs to be stored to env to be able to construct the
-        # refuris later
-        poi_info = {
-            'docname': env.docname,
-        }
-        if 'previous' in self.options:
-            poi_info['previous'] = self.options['previous']
-        if 'next' in self.options:
-            poi_info['next'] = self.options['next']
-
-        env.poi_all[name] = poi_info
+        if not 'not_in_book' in self.options:
```

Comment:
Instead of putting so much code into a simple if statement, it is better to check the condition and return early from the run method. Then, there is no need to indent so many existing lines.

For example, I think you could do this right after the `node` variable has been defined:

```
if 'not_in_book' in self.options:
    poi_info = {
        'docname': env.docname,
    }
    env.poi_all[name] = poi_info
    return [node]
```